### DISCUSSION TITLE
Access Action's form state in modalContentFooter() view?

### DISCUSSION THREAD

**Discussion by alexpgates** [filamentphp/filament#13055] [Help] [answered]
### Package

Actions

### Package Version

v3.2.85

### How can we help you?

How can I access the state of an actions `->form()` inside the modal content?

I'm trying to create an action modal (slideover) like this: 

```
Action::make('review')
    ->form([
        Select::make('coverage_amount')
            ->label('Amount')
            ->options([
                '100000' => '100,000',
                '200000' => '200,000',
                '300000' => '300,000',
            ])
            ->placeholder('Select coverage amount')
            ->required()
    ])
    ->modalContentFooter(fn (CourtCase $record): View => view(
        'filament.actions.elect-coverage',
        ['record' => $record],
    ))
    ->action(function (CourtCase $record, $data) {
        dd($data); 
    })
    ->slideOver(),
```

I want the user to be able to select different coverage amounts and see the calculated fees. **How can I access the value of `coverage_amount` from within the `modalContentFooter()`?** 

Is that possible?

**Comment by alexpgates** [comment]
Is the right approach to not attempt to show dynamic data in the modalContentFooter and instead use a Livewire::make() component instead? At least that's within the same form, right? Can I use `$get` or `$set` ?? Or can I get the data in my component some other way? 

```
Action::make('Test')
    ->form([
        Select::make('coverage_amount')
            ->label('Amount')
            ->options(fn (CourtCase $case) => $case->getCoverageOptions())
            ->default(fn (CourtCase $case) => $case->defaultCoverage())
            ->placeholder('Select coverage amount')
            ->afterStateUpdated(function (Set $set, ?string $state) {
                $set('SOMETHING IN THE LIVEWIRE COMPONENT', $state);
            })
            ->live(),
        Livewire::make(PolicyPreview::class),
    ])
    ->action(function (CourtCase $case, $data) {
        dd($data);
    }),
```

Any guidance in any direction would be incredibly helpful. The bottom line is, I'm trying to create an action modal from a table that has a select dropdown but I want to be able to display some additional calculations in the modal based off the current selection. 

This works to get the default state from the select field to the livewire component, but it isn't reactive. 

```
->form([
    Select::make('coverage_amount')
        ->label('Amount')
        ->options(fn (CourtCase $case) => $case->getCoverageOptions())
        ->default(fn (CourtCase $case) => $case->firmRequestedDefaultCoverage())
        ->placeholder('Select coverage amount')
        ->live(),
    Livewire::make(PolicyPreview::class)
        ->data(fn (Get $get) => [
            'amount' => $get('coverage_amount'),
        ]),
])
```

**Comment by leandrocfe** [comment]
Is it the same question in the Discord?
https://discord.com/channels/883083792112300104/1246657725983752303/1246896148548157583

**Reply by alexpgates** [reply]
Yes indeed. Thank you!

**Comment by alexpgates** [comment, marked as answer]
The solution to this was to use a `ViewField` - this allowed me to use the state of one form field in a blade view to display modal content based on the form field selection. 

Thank you to @leandrocfe for the assistance. 

```php
ViewField::make('amount')
->view('policy-preview')
```

```html
<!-- resources/views/policy-preview.blade.php -->
<div>
    {{ $getState() }}
</div>
```

And use afterStateUpdated in the select to set the value

```php
->afterStateUpdated(function (Set $set, ?string $state) {
    $set('amount', $state);
})
```
